table/block_based/block_based_table_builder.cc

```diff
@@ -1008,6 +1018,7 @@ void BlockBasedTableBuilder::WriteBlock(BlockBuilder* block,
     assert(is_data_block);
     rep_->data_block_buffers.emplace_back(std::move(raw_block_contents));
     rep_->data_begin_offset += rep_->data_block_buffers.back().size();
+    ReserveOneBufferedDataBlockMemWithCache();
```

Comment:
Question 5:
As for the practice of using compiler flag such NDEBUG, ROCKSDB_LITE for new feature
(1)how to tell whether I need to use NDEBUG for a section of code?
(2)for ROCKSDB_LITE, I read through https://github.com/facebook/rocksdb/blob/master/ROCKSDB_LITE.md but I still can't tell whether I need to use ROCKSDB_LITE or not - need some help here